tests/santa/test_santa_api_views.py

```diff
@@ -724,13 +727,51 @@ def test_eventupload_bundle_binary(self, post_event):
     # postflight
 
     def test_rule_postflight_not_enrolled(self):
-        url = reverse("santa:postflight", args=(self.enrollment_secret.secret, uuid.uuid4()))
+        url = reverse("santa_public:postflight", args=(self.enrollment_secret.secret, uuid.uuid4()))
         response = self.post_as_json(url, {})
         self.assertEqual(response.status_code, 403)
 
     def test_postflight(self):
-        url = reverse("santa:postflight", args=(self.enrollment_secret.secret, self.enrolled_machine.hardware_uuid))
+        url = reverse("santa_public:postflight", args=(self.enrollment_secret.secret, self.enrolled_machine.hardware_uuid))
         response = self.post_as_json(url, {})
         self.assertEqual(response.status_code, 200)
         json_response = response.json()
         self.assertEqual(json_response, {})
+
+    def test_legacy_public_urls_are_disabled_on_tests(self):
+        hardware_uuid = self._get_preflight_data()
+        routes = [
+            ('preflight', (get_random_string(12), hardware_uuid)),
+            ('ruledownload', (self.enrollment_secret.secret, uuid.uuid4())),
+            ('eventupload', (self.enrollment_secret.secret, uuid.uuid4())),
+            ('postflight', (self.enrollment_secret.secret, uuid.uuid4())),
+        ]
+
+        for route, args in routes:
+            with self.assertRaises(NoReverseMatch):
+                reverse(f"santa_public_legacy:{route}", args=args)
+            self.assertIsNotNone(reverse(f"santa_public:{route}", args=args))
+
+    def test_mount_legacy_public_endpoints_flag_is_working(self):
+        hardware_uuid = self._get_preflight_data()
+        routes = [
+            ('preflight', (get_random_string(12), hardware_uuid)),
+            ('ruledownload', (self.enrollment_secret.secret, uuid.uuid4())),
+            ('eventupload', (self.enrollment_secret.secret, uuid.uuid4())),
+            ('postflight', (self.enrollment_secret.secret, uuid.uuid4())),
+        ]
+        url_prefix = "/public"
+
+        santa_conf = settings._collection["apps"]._collection["zentral.contrib.santa"]
+        santa_conf._collection["mount_legacy_public_endpoints"] = True
+        urlpatterns_w_legacy = tuple(build_urlpatterns_for_zentral_apps())
+        santa_conf._collection["mount_legacy_public_endpoints"] = False
+        urlpatterns_wo_legacy = tuple(build_urlpatterns_for_zentral_apps())
+
+        for route, args in routes:
+            self.assertEqual(
+                reverse(f"santa_public:{route}", urlconf=urlpatterns_w_legacy, args=args),
+                url_prefix + reverse(f"santa_public_legacy:{route}", urlconf=urlpatterns_w_legacy, args=args)
+            )
+            with self.assertRaises(NoReverseMatch):
+                reverse("santa_public:{route}", urlconf=urlpatterns_wo_legacy, args=args)
```

Comment:
it should be the santa_legacy_public namespace failing. it is failing because the f of the f-string is missing.

Reply:
Also, the url_prefix doesn't have to be a variable.